frontend/src/components/GrantAccessForm.vue

```diff
@@ -0,0 +1,183 @@
+                        <select id="roleSelect" name="role" class="form-select" aria-label="Role Select"
+                            v-model="formData.role"
+                            :class="v$.role.$dirty ? v$.role.$invalid ? 'is-invalid' : 'is-valid' : null"
+                            @change.native="v$.role.$touch()" @blur.native="v$.role.$touch()">
+                            <option v-for="role in applicationRoleOptions" :value="role">{{ role.role_name }}</option>
+                        </select>
+                        <div class="text-danger" v-for="error of v$.role.$errors" :key="error.$uid">
+                            <small>{{ error.$message }}</small>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div class="row" v-if="formData.role?.role_name == 'A'">
+                <div class="form-group col-md-3">
+                    <label for="forestClientInput" class="control-label">Forest Client</label>
+                    <input type="text" id="forestClientInput" class="form-control"
+                        :maxlength="FOREST_CLIENT_INPUT_MAX_LENGTH" placeholder="Forest Client Id - 8 digits"
+                        v-model="formData.forestClientNumber" v-on:keypress="onlyDigit($event)">
+                </div>
+            </div>
+
+            <div class="row gy-3">
+                <div class="col-auto">
+                    <button type="submit" id="grantAccessSubmit" class="btn btn-primary mb-3"
+                        :disabled="!v$.$anyDirty || v$.$invalid">
+                        Grant Access
+                    </button>
+                </div>
+            </div>
+        </form>
+    </div>
+</template>
+
+<script lang="ts">
+import { ApiService, type ApplicationRoleResponse, type GrantUserRoleRequest } from '@/services/ApiService';
+import { selectedApplication } from '@/services/ApplicationState';
+import { ref } from 'vue';
+import { useToast } from 'vue-toastification';
+
+import { reactive } from 'vue';
+import { useVuelidate } from '@vuelidate/core';
+import { required, minLength, maxLength } from '@vuelidate/validators';
+import type { button } from 'aws-amplify';
+
+export default {
+    setup() {
+        const FOREST_CLIENT_INPUT_MAX_LENGTH = 8;
+        const domainOptions = { IDIR: 'I', BCEID: 'B' }; // TODO, load it from backend when backend has the endpoint.
+
+        const apiService = new ApiService();
+
+        const defaultFormData = {
+            domain: domainOptions.BCEID,
+            userId: null,
+            forestClientNumber: null,
+            role: null as unknown as ApplicationRoleResponse
+        };
+
+        const formData = reactive({ ...defaultFormData });
+        const rules = {
+            domain: {},
+            userId: { required, minLength: minLength(3), maxLength: maxLength(20) },
```

Comment:
Not sure what the actual min/max should be, but this is a good example of form validation at least.

Reply:
Max length was already a validation parameter, the min length can be adjusted to whatever it is required. Yes, I used to showcase some of the validations possible. It is slightly different in vee-validate but the concept is the same.